soh/src/overlays/actors/ovl_Bg_Mori_Kaitenkabe/z_bg_mori_kaitenkabe.c

```diff
@@ -93,19 +93,38 @@ void BgMoriKaitenkabe_Wait(BgMoriKaitenkabe* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
 
     if (this->dyna.unk_150 > 0.001f) {
-        this->timer++;
-        if ((this->timer > (28 - CVarGetInteger(CVAR_ENHANCEMENT("FasterBlockPush"), 0) * 4)) && !Player_InCsMode(play)) {
-            BgMoriKaitenkabe_SetupRotate(this);
-            func_8002DF54(play, &this->dyna.actor, 8);
-            Math_Vec3f_Copy(&this->lockedPlayerPos, &player->actor.world.pos);
-            push.x = Math_SinS(this->dyna.unk_158);
-            push.y = 0.0f;
-            push.z = Math_CosS(this->dyna.unk_158);
-            leverArm.x = this->dyna.actor.world.pos.x - player->actor.world.pos.x;
-            leverArm.y = 0.0f;
-            leverArm.z = this->dyna.actor.world.pos.z - player->actor.world.pos.z;
-            BgMoriKaitenkabe_CrossProduct(&torque, &push, &leverArm);
-            this->rotDirection = (torque.y > 0.0f) ? 1.0f : -1.0f;
+        // #region SOH [Enhancement]
+        if (CVarGetInteger(CVAR_ENHANCEMENT("FasterBlockPush"), 0)) {
+            this->timer++;
+            if ((this->timer > (28 - CVarGetInteger(CVAR_ENHANCEMENT("FasterBlockPush"), 0) * 4)) &&
+                !Player_InCsMode(play)) {
+                BgMoriKaitenkabe_SetupRotate(this);
+                Math_Vec3f_Copy(&this->lockedPlayerPos, &player->actor.world.pos);
+                push.x = Math_SinS(this->dyna.unk_158);
+                push.y = 0.0f;
+                push.z = Math_CosS(this->dyna.unk_158);
+                leverArm.x = this->dyna.actor.world.pos.x - player->actor.world.pos.x;
+                leverArm.y = 0.0f;
+                leverArm.z = this->dyna.actor.world.pos.z - player->actor.world.pos.z;
+                BgMoriKaitenkabe_CrossProduct(&torque, &push, &leverArm);
+                this->rotDirection = (torque.y > 0.0f) ? 1.0f : -1.0f;
+            }
+            // #endregion
+        } else {
+            this->timer++;
+            if (this->timer > 28 && !Player_InCsMode(play)) {
+                BgMoriKaitenkabe_SetupRotate(this);
+                func_8002DF54(play, &this->dyna.actor, 8);
+                Math_Vec3f_Copy(&this->lockedPlayerPos, &player->actor.world.pos);
+                push.x = Math_SinS(this->dyna.unk_158);
+                push.y = 0.0f;
+                push.z = Math_CosS(this->dyna.unk_158);
+                leverArm.x = this->dyna.actor.world.pos.x - player->actor.world.pos.x;
+                leverArm.y = 0.0f;
+                leverArm.z = this->dyna.actor.world.pos.z - player->actor.world.pos.z;
+                BgMoriKaitenkabe_CrossProduct(&torque, &push, &leverArm);
+                this->rotDirection = (torque.y > 0.0f) ? 1.0f : -1.0f;
+            }
```

Comment:
```suggestion
        this->timer++;
        u8 timerThreshold = 28 - CVarGetInteger(CVAR_ENHANCEMENT("FasterBlockPush"), 0) * 4;
        if ((this->timer > timerThreshold && !Player_InCsMode(play)) {
            BgMoriKaitenkabe_SetupRotate(this);
            if (!CVarGetInteger(CVAR_ENHANCEMENT("FasterBlockPush"), 0)) {
                func_8002DF54(play, &this->dyna.actor, 8);
            }
            Math_Vec3f_Copy(&this->lockedPlayerPos, &player->actor.world.pos);
            push.x = Math_SinS(this->dyna.unk_158);
            push.y = 0.0f;
            push.z = Math_CosS(this->dyna.unk_158);
            leverArm.x = this->dyna.actor.world.pos.x - player->actor.world.pos.x;
            leverArm.y = 0.0f;
            leverArm.z = this->dyna.actor.world.pos.z - player->actor.world.pos.z;
            BgMoriKaitenkabe_CrossProduct(&torque, &push, &leverArm);
            this->rotDirection = (torque.y > 0.0f) ? 1.0f : -1.0f;
        }
```

Reply:
I wonder if it would be better to put all of this behind an "apply faster block push to forest pillars" and then include both items in the tooltip? It feels half-baked to only have them move faster, but agree that it's distinct.

For the organization, I agree in-line is a little easier to read (it's usually how I start off), but I was trying to follow the contribution guide with the regions/blocking out vanilla code - not sure if that is still the standard! Happy to switch to whatever format we need.